src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -866,29 +964,54 @@ impl SimulationElement for A320PowerTransferUnitController {
         self.eng_2_master_on = state.read_bool("GENERAL ENG STARTER ACTIVE:2");
         self.weight_on_wheels = state.read_bool("SIM ON GROUND");
     }
+
+    fn receive_power(&mut self, supplied_power: &SuppliedPower) {
+        self.is_powered = supplied_power.potential_of(&self.powering_bus).is_powered();
+    }
 }
 
 struct A320RamAirTurbineController {
+    is_solenoid_1_powered: bool,
+    solenoid_1_bus: ElectricalBusType,
+
+    is_solenoid_2_powered: bool,
+    solenoid_2_bus: ElectricalBusType,
+
     should_deploy: bool,
     eng_1_master_on: bool,
     eng_2_master_on: bool,
 }
 impl A320RamAirTurbineController {
+    const POWER_BUS_SOLENOID_1: ElectricalBusType = ElectricalBusType::DirectCurrentHot(1);
+    const POWER_BUS_SOLENOID_2: ElectricalBusType = ElectricalBusType::DirectCurrentHot(2);
+
     fn new() -> Self {
         Self {
+            is_solenoid_1_powered: false,
+            solenoid_1_bus: Self::POWER_BUS_SOLENOID_1,
+
+            is_solenoid_2_powered: false,
+            solenoid_2_bus: Self::POWER_BUS_SOLENOID_2,
+
             should_deploy: false,
             eng_1_master_on: false,
             eng_2_master_on: false,
         }
     }
 
-    fn update(&mut self, context: &UpdateContext) {
-        // RAT Deployment
-        // Todo check all other needed conditions this is faked with engine master while it should check elec buses
-        self.should_deploy = !self.eng_1_master_on
-            && !self.eng_2_master_on
-            // Todo get speed from ADIRS
-            && context.indicated_airspeed() > Velocity::new::<knot>(100.)
+    fn update(
+        &mut self,
+        overhead_panel: &A320HydraulicOverheadPanel,
+        emergency_electrical_overhead: &A320EmergencyElectricalOverheadPanel,
+        is_in_emergency_elec: bool,
+    ) {
+        let should_solenoid_1_deploy_if_powered = overhead_panel.rat_man_on_push_button_is_on();
+
+        let should_solenoid_2_deploy_if_powered =
+            is_in_emergency_elec || emergency_electrical_overhead.rat_and_emer_gen_man_on();
```

Comment:
**suggestion**: Rename to `solenoid_1_should_trigger_deploy(ment)`; as the current name suggests the solenoid is deployed, not the RAT.

Reply:
**thought**: Maybe the solenoids should have their own type, so they can handle their output and being powered internally?